analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -312,6 +316,40 @@ public void ProjectConfiguration_WhenInvalidXml_ThrowException()
                 .WithMessage("File SonarProjectConfig.xml has been added as an AdditionalFile but could not be read and parsed.");
         }
 
+        [TestMethod]
+        public void IsAnalysisScopeMatching_NoCompilation_IsMatching() =>
+            SonarAnalysisContext.IsAnalysisScopeMatching(null, true, null).Should().BeTrue();
+
+        [DataTestMethod]
+        [DataRow(true, false, MainTag)]
+        [DataRow(true, false, MainTag, TestTag)]
+        [DataRow(true, true, TestTag)]
+        [DataRow(true, true, MainTag, TestTag)]
+        [DataRow(false, false, TestTag)]
+        [DataRow(false, false, TestTag, TestTag)]
+        [DataRow(false, true, MainTag)]
+        [DataRow(false, true, MainTag, MainTag)]
+        public void IsAnalysisScopeMatching_SingleDiagnostis_WithOneOrMoreScopes(bool expectedResult, bool isTestProject, params string[] singleDiagnosticTags)
```

Comment:
I'd put expectedResult at the end, for consistency (input, output) like in `public void IsTestProject_Standalone(ProjectType projectType, bool expectedResult)`

Reply:
and rename `singleDiagnosticTags` to `testRuleScope`

Reply:
I have it intentionally at the beginning to have table-like structure (easier to read), as the argument count varies. And I couldn't use `params` argument like that (more syntax noise with worse readability).